### PR TITLE
Restructure DETR post-processing, return prediction scores

### PR DESCRIPTION
**Pull request by alaradirik**
# What does this PR do?
- Restructures `DetrFeatureExtractor.post_process_instance_segmentation` and `DetrFeatureExtractor.post_process_panoptic_segmentation` methods, adds `score` key to output dictionaries
- Updates feature extractor tests

## Before submitting
- [ X] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [X ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
     This was discussed on [Notion](https://www.notion.so/Post-processing-of-vision-models-33a8165333144581862014d83e96191b?d=462e95b396fc4ea49ffa602146bd91f5#d152a0d70b4c4cf599f642e9b891dba1).
- [ X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
LGTM!

File: src/transformers/models/detr/feature_extraction_detr.py [hunk 1]
```diff
@@ -141,6 +141,26 @@ def binary_mask_to_rle(mask):
     return [x for x in runs]
 
 
+def convert_segmentation_to_rle(segmentation):
+    """
+    Args:
+    Converts given segmentation map of shape (height, width) to the run-length encoding (RLE) format.
+        segmentation (`torch.Tensor` or `numpy.array`):
+            A segmentation map of shape `(height, width)` where each value denotes a segment or class id.
```
Comment (NielsRogge) [inline src/transformers/models/detr/feature_extraction_detr.py:149]:
```suggestion
    Converts given segmentation map of shape (height, width) to the run-length encoding (RLE) format.
    
    Args:
            segmentation (`torch.Tensor` or `numpy.array`):
                     A segmentation map of shape `(height, width)` where each value denotes a segment or class id.
```

Comment (NielsRogge) [reply src/transformers/models/detr/feature_extraction_detr.py:149]: Sorry the tab is a bit messed up on the Github web UI, but you get the point

Comment (alaradirik) [reply src/transformers/models/detr/feature_extraction_detr.py:149]: No worries, fixing it now!

Comment (alaradirik) [reply src/transformers/models/detr/feature_extraction_detr.py]: @NielsRogge could you take a look when you have the time? I'll merge the PR is everything is OK.

**Pull request merged**